src/iop/hazeremoval.c

```diff
@@ -273,18 +288,20 @@ static void transition_map(const const_rgb_image img1, const gray_image img2, co
                            const float strength)
 {
   const size_t size = (size_t)img1.height * img1.width;
+  const float *const restrict in_data = img1.data;
+  float *const restrict out_data = img2.data;
+  const dt_aligned_pixel_t A0_inv = { 1.0f / A0[0], 1.0f / A0[1], 1.0f / A0[2], 1.0f };
 #ifdef _OPENMP
-#pragma omp parallel for default(none) \
-  dt_omp_firstprivate(A0, img1, img2, size, strength) \
-  schedule(static)
+#pragma omp parallel for simd aligned(in_data, out_data: 64) default(none) \
+  dt_omp_firstprivate(A0_inv, in_data, out_data, size, strength) \
+  schedule(simd:static)
 #endif
   for(size_t i = 0; i < size; i++)
   {
-    const float *pixel = img1.data + i * img1.stride;
-    float m = pixel[0] / A0[0];
-    m = fminf(pixel[1] / A0[1], m);
-    m = fminf(pixel[2] / A0[2], m);
-    img2.data[i] = 1.f - m * strength;
+    const float *pixel = in_data + 4*i;
+    float m = MIN(pixel[0] * A0_inv[0], pixel[1] * A0_inv[1]);
```

Comment:
`const float m = ...`

Possibly with the 2 MIN if you agree with my proposal above.